packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1496,6 +1502,203 @@ class SliverAppBar extends StatefulWidget {
        assert(stretchTriggerOffset > 0.0),
        assert(collapsedHeight == null || collapsedHeight >= toolbarHeight, 'The "collapsedHeight" argument has to be larger than or equal to [toolbarHeight].');
 
+  /// Creates a Material Design app bar matching the 'Medium' type
+  /// that can be placed in a [CustomScrollView].
+  ///
+  /// By default, it configures a [SliverAppBar] with the appropriate defaults
+  /// for the 'Medium' type Material 3 app bar. It starts fully expanded with
+  /// the title in an area underneath the main row of icons. When the
+  /// [CustomScrollView] is scrolled, this title will be scrolled under the
+  /// main row. When it is fully collapsed, a smaller version of the title
+  /// will fade in on the main row. The reverse will happen if it is expanded
+  /// again.
+  ///
+  /// {@tool dartpad}
+  /// This sample shows how to use [SliverAppBar.medium] in a [CustomScrollView].
+  ///
+  /// ** See code in examples/api/lib/material/app_bar/sliver_app_bar.2.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///   * [AppBar], for the 'Small' and 'Center-aligned' types of app bar.
+  ///   * [SliverAppBar.large], for the 'Large' type of app bar.
+  ///   * https://m3.material.io/components/top-app-bar/overview, the Material 3
+  ///     app bar specification.
+  factory SliverAppBar.medium({
+    Key? key,
+    Widget? leading,
+    bool automaticallyImplyLeading = true,
+    Widget? title,
+    List<Widget>? actions,
+    Widget? flexibleSpace,
+    PreferredSizeWidget? bottom,
+    double? elevation,
+    double? scrolledUnderElevation,
+    Color? shadowColor,
+    Color? surfaceTintColor,
+    Color? backgroundColor,
+    Color? foregroundColor,
+    IconThemeData? iconTheme,
+    IconThemeData? actionsIconTheme,
+    bool primary = true,
+    bool? centerTitle,
+    bool excludeHeaderSemantics = false,
+    double? titleSpacing,
+    double collapsedHeight = _MediumScrollUnderFlexibleConfig.collapsedHeight,
+    double expandedHeight = _MediumScrollUnderFlexibleConfig.expandedHeight,
+    bool floating = false,
+    bool pinned = true,
+    bool snap = false,
+    bool stretch = false,
+    double stretchTriggerOffset = 100.0,
+    AsyncCallback? onStretchTrigger,
+    ShapeBorder? shape,
+    double toolbarHeight = _MediumScrollUnderFlexibleConfig.collapsedHeight,
+    double? leadingWidth,
+    TextStyle? toolbarTextStyle,
+    TextStyle? titleTextStyle,
+    SystemUiOverlayStyle? systemOverlayStyle,
+  }) {
+    return SliverAppBar(
+      key: key,
+      leading: leading,
+      automaticallyImplyLeading: automaticallyImplyLeading,
+      actions: actions,
+      flexibleSpace: flexibleSpace ?? _ScrollUnderFlexibleSpace(
+        title: title,
```

Comment:
missing `variant: _ScrollUnderFlexibleVariant.medium,`?

Reply:
This was the default, so wasn't needed (and we have a lint that would complain if we specified it). But to make this more obvious, I removed the default and made it a required param. Thx.